### PR TITLE
Add 'load' & chart options

### PR DESCRIPTION
**Pull request by tippenein**
I really like this interface for charts! This Pull Request is sort of meant as an RFP or just a way of discussing what direction the library could go.

The style and choices I made implementing these are dubious at best, but I'd like to figure out how best to move this forward.

What's implemented here:

- Gauge charts get a GaugeOpts to put config options in. (The rest of the charts could follow this example with their own specifics)
- `load :: Chart -> Datum -> Chart` for loading in new columns
- unload _could_ be added.. maybe the type is Chart -> Text -> Chart where the 2nd arg describes the data index to unload? I'm not sure of C3's usage of unload beyond the examples I've seen.

Anyway, thanks for starting this library

### CODE REVIEW

**Comment by tippenein** [comment]
I introduced a type called `Datum` which can encompass Column, Row, etc. There's one gross bit I hacked around, because I couldn't figure out how to encode the datum with the other base chart data (https://github.com/mbeidler/ghcjs-c3/pull/1/files#diff-4e8099f8f4b1ab8aee0d77636d3652a3R109)

Actually, the whole ChartData instance is pretty bad.

I'll keep chunking away on this and feel free to pour in your displeasure (and possibly suggestions ;) ) at this. 
:sunny:

File: src/C3/Types.hs [hunk 1]
```diff
@@ -114,8 +136,3 @@ instance ToJSON ChartType where
   toJSON Pie        = String "pie"
   toJSON Donut      = String "donut"
   toJSON (Gauge g)  = object [ "gauge" .= toJSON g]
```
Comment (mbeidler) [inline src/C3/Types.hs]: Doesn't this produce JSON like `{ "data": { "type": { "gauge": { "min": 0 } } } }`? The example I saw at http://c3js.org/samples/chart_gauge.html shows the gauge options at the same level as `data` in the JSON. That was where I put it in the code sample above.

File: src/C3/Types.hs [hunk 2]
```diff
@@ -82,18 +102,20 @@ instance ToJSON ChartSizeOptions where
...
 instance ToJSON ChartData where
-  toJSON ChartData{..} = object [
-    "type"    .= dumbType chartType,
-    "columns" .= chartColumns ]
+  toJSON ChartData{..} = object conjoined
     where
+      base = [ "type"    .= dumbType chartType ]
+      --XXX would be nice to `toJSON chartDatum` and combine with base
+      conjoined =  base ++ datum
+      datum = case chartDatum of
```
Comment (tippenein) [inline src/C3/Types.hs]: This works nicely. I added the `{size: {height: <some_int>}}` option at that level as well

File: src/C3/Types.hs [hunk 3]
```diff
@@ -26,6 +33,14 @@ data ChartData = ChartData
...
   }
 
+data GaugeOpts
+  = GaugeOpt
+  { _gauge_min :: Int
+  , _gauge_max :: Int
+  , _gauge_units :: Text
+  , _gauge_width :: Int
+  } deriving Show
+
```
Comment (tippenein) [inline src/C3/Types.hs]: Yeah, I should've been consistent

**Comment by tippenein** [comment]
I feel like we could do better than `Datum`. For example, the way to make a Row for a Pie chart is pretty gross.
```haskell
pieData = Rows ["US", "Them", "something", "Gary Busey"] [[10],[5],[4],[8]]
```
It can cause unraised errors (exactly what `fe91c6a` was fixing)

Perhaps some sort of differentiator for data which doesn't have an X and Y axis? Something like a Key/Value pairing when applicable?

**Comment by mbeidler** [comment]
Sorry to take so long on this. I agree that the list of lists approach is unsatisfactory. Does this serve a different purpose than the existing column-based API? I was thinking that it might be useful to have combinators for multi-dimensional data (where there are additional constraints, e.g. that an X and Y column must have the same number of samples). Those combinators could have stronger types to give a nicer API.

**Comment by tippenein** [comment]
I was just aiming for consistency with the C3 api, but I don't think Row adds anything which can't be handled by `Column` at the moment

**Comment by tippenein** [comment]
What do you think about merging this? `Gauge` is mostly complete and relatively good as a pattern for the rest. We can figure out better `Datum` types in a later PR. I'm in no hurry, but for the sake of _focused_ PRs

**Comment by mbeidler** [comment]
I've merged your PR. Thanks!

**Pull request merged**